src/components/SingleBook.js

```diff
@@ -1,23 +1,44 @@
 import React from 'react';
 import PropTypes, { string } from 'prop-types';
+import { useDispatch } from 'react-redux/es/exports';
+import { removeBookActionCreator } from '../redux/books/books';
 
 const SingleBook = (props) => {
-  const { book } = props;
+  const dispatch = useDispatch();
+  const { index, book } = props;
   const { title, author } = book;
+  const onClickHandle = () => {
+    dispatch({
+      type: removeBookActionCreator().type,
+      index,
+    });
```

Comment:
- [ ] One of the main purpose of the book id property is to implement book deletion. Kindly implement your book deletion using the book id propoerty.